ssl/t1_lib.c

```diff
@@ -2302,14 +2399,9 @@ int tls_choose_sigalg(SSL *s, int fatalerrs)
                 || lu->sig == EVP_PKEY_DSA
                 || lu->sig == EVP_PKEY_RSA)
                 continue;
-            if (!tls1_lookup_md(lu, NULL))
+            /* Check that we have a cert, and signature_algorithms_cert */
+            if (!tls1_lookup_md(lu, NULL) || !has_usable_cert(s, lu, -1))
                 continue;
-            if (!ssl_has_cert(s, lu->sig_idx)) {
-                if (lu->sig_idx != SSL_PKEY_RSA_PSS_SIGN
-                        || !ssl_has_cert(s, SSL_PKEY_RSA))
-                    continue;
```

Comment:
I'm not seeing this original logic reflected in `has_usable_cert()`. Shouldn't it be?

Reply:
You are correct that it is not reflected in has_usable_cert().  It is also correct for this logic to not be reflected in has_usable_cert().  This code is in the ```SSL_IS_TLS13()``` block, but it represents the compatibility fallback wherein we let rsaEncryption keys generate PSS signatures.  The whole point of the new codepoints in draft-23 is that we should not do this fallback, since we now have separate codepoints for "PSS signature using rsaEncryption key" and "PSS signature using PSS key".  I left this fallback in for the TLS 1.2 case since I thought there was a way for an old client that only does TLS 1.2 to be trying the PSS signature but not know about the split codepoints.  But maybe that's not actually true so that logic should also be removed...